src/cljs/rems/application.cljs

```diff
@@ -1178,35 +1177,37 @@
                (when-let [missing-duos (seq (unmatched-duos application-duo-matches))]
                  [:div.alert.alert-warning
                   [:p (text :t.applications.duos.validation/missing-duo-codes)]
-                  [:ul (for [duo missing-duos
-                             :let [label (localized (:duo/label duo))]]
-                         ^{:key (:duo/id duo)}
-                         [:li (text-format :t.label/dash
-                                           (:duo/shorthand duo) label)])]])
+                  (into [:ul]
+                        (for [duo missing-duos]
```

Comment:
This is not a very satisfying solution. In some projects I've made some custom `for` loops for components to clean up the code. Another solution would be to, instead of `:ul` to have our own `list` component that would accept the sequence (or multiple sequences to get rid of extra `concat` etc.) and do the needful.

E.g., `[List (for [duo missing-duos] ...)]`

We could consider doing such cleanup after we merge the open PRs and do a release.